File: dotnet/cafdemalihapa/Applications/ApplicationSwitcher.cs
using CafdemalihapaApp = cafdemalihapa.Applications.Cafdemalihapa.Cafdemalihapa;
using cafdemalihapa.Applications.NotepadPlusPlusFileManagement;

namespace cafdemalihapa.Applications
{
    public class ApplicationSwitcher(
        CafdemalihapaApp cafdemalihapa,
        NotepadPlusPlusFileManagementCommandSwitcher notepadPlusPlusFileManagementCommandSwitcher,
        IdeManagement.IdeManagement ideManagement,
        DirectoryManagement.DirectoryManagement directoryManagement
        )
    {
        public void Run()
        {
            if (Applications.Get().IsCafdemalihapa())
            {
                cafdemalihapa.Run();
            }

            if (Applications.Get().IsNotepadPlusPlusFileManagementApplication())
            {
                notepadPlusPlusFileManagementCommandSwitcher.Run();
            }

            if (Applications.Get().IsIdeManagementApplication())
            {
                ideManagement.Run();
            }
            directoryManagement.Run();
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/ApplicationSwitcher.fs
namespace cafdemwimu.console.Applications

open cafdemwimu.console.Applications.NotepadPlusPlusFileManagement

type CafdemalihapaApp = cafdemwimu.console.Applications.Cafdemalihapa.Cafdemalihapa

type ApplicationSwitcher
    (
        cafdemalihapa: CafdemalihapaApp,
        notepadPlusPlusFileManagementCommandSwitcher: NotepadPlusPlusFileManagementCommandSwitcher,
        ideManagement: cafdemwimu.console.Applications.IdeManagement.IdeManagement,
        directoryManagement: cafdemwimu.console.Applications.DirectoryManagement.DirectoryManagement
    ) =
    member _.Run() =
        if Applications.Get().IsCafdemalihapa() then
            cafdemalihapa.Run()

        if Applications.Get().IsNotepadPlusPlusFileManagementApplication() then
            notepadPlusPlusFileManagementCommandSwitcher.Run()

        if Applications.Get().IsIdeManagementApplication() then
            ideManagement.Run()

        directoryManagement.Run()
